server/server.go

```diff
@@ -407,6 +408,53 @@ func (server HTTPServer) readAggregatorReportForClusterList(
 	return &aggregatorResponse, true
 }
 
+func (server HTTPServer) readAggregatorReportForClusterListFromBody(
+	orgID types.OrgID, request *http.Request, writer http.ResponseWriter,
+) (*types.ClusterReports, bool) {
+	aggregatorURL := httputils.MakeURLToEndpoint(
+		server.ServicesConfig.AggregatorBaseEndpoint,
+		ira_server.ReportForListOfClustersPayloadEndpoint,
+		orgID,
+	)
+
+	body, err := ioutil.ReadAll(request.Body)
+	if err != nil {
+		handleServerError(writer, err)
+		return nil, false
+	}
+	// #nosec G107
+	aggregatorResp, err := http.Post(aggregatorURL, "application/json", bytes.NewBuffer(body))
+	if err != nil {
+		handleServerError(writer, err)
+		return nil, false
+	}
+
```

Comment:
probably will be better to split the function at this line. The reason - free the `body` + associated `Buffer` as fast as possible as it is memory intensive.

Reply:
Makes sense. Updating